### PR TITLE
Add new traits to apply the filter only to branches, pull requests or tags

### PR DESCRIPTION
**Pull request by Trucosuso**
Add three new tags to apply the filter only to one type of reference, branch, pull request or tag. These traits can be combined in any way to apply the filter only to the desired kind of references. The README of the plugin has been updated with a short description of the new traits.

As the filter for each type of ref was being used two times I moved it to a separate package in order to help with maintenance.

This work originated from https://github.com/jenkinsci/scm-filter-aged-refs-plugin/pull/45#issuecomment-2425974298

### Testing done

I lack the knowledge to add proper automatic testing, but this feature has been tested manually on multiple Bitbucket and GitHub repositories with a different amount of branches, pull requests and tags with great results.

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: scm-filter-aged-refs-common/src/main/java/org/jenkinsci/plugins/scm_filter/AgedTypeRefsTrait.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        @Override
+        @NonNull
+        public abstract String getDisplayName();
+
+        public abstract String getRefName();
+
+        @Restricted(NoExternalUse.class)
+        @POST
+        public FormValidation doCheckRetentionDays(@QueryParameter String value) {
```
[the remark this answers is not in the document]
Comment (Trucosuso) [reply scm-filter-aged-refs-common/src/main/java/org/jenkinsci/plugins/scm_filter/AgedTypeRefsTrait.java:51]: I'm not sure about what causes this

**Comment by piomis** [comment]
I'm really looking forward to see this change merged and released :)

By the way is there any chance to have it also extended with some kind of "exlusion" filter (e.g. I would like to exclude master/main branches from being deleted even if there was no recent activity).

File: scm-filter-aged-refs-common/src/main/resources/org/jenkinsci/plugins/scm_filter/AgedTypeRefsTrait/help.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<div>
+    Defines the number of days to accept an unmodified git ref as candidate for being built (supposing it contains a Jenkinsfile).
+    In a practical effect it is a ripoff from the Orphaned Item Strategy but on indexing time.
```
[the remark this answers is not in the document]
Comment (Trucosuso) [reply scm-filter-aged-refs-common/src/main/resources/org/jenkinsci/plugins/scm_filter/AgedTypeRefsTrait/help.html:3]: Sure! I also find it weird to have it there 😄 
Should I also remove it from `scm-filter-aged-refs-common/src/main/resources/org/jenkinsci/plugins/scm_filter/AgedRefsTrait/help.html`?

[the remark this answers is not in the document]
Comment (TobiX) [reply scm-filter-aged-refs-common/src/main/resources/org/jenkinsci/plugins/scm_filter/AgedTypeRefsTrait/help.html:3]: Yeah, should be fine to include that in this PR.

[the remark this answers is not in the document]
Comment (Trucosuso) [reply scm-filter-aged-refs-common/src/main/resources/org/jenkinsci/plugins/scm_filter/AgedTypeRefsTrait/help.html:3]: Ok! I'll do that

File: bitbucket-scm-filter-aged-refs/src/main/java/org/jenkinsci/plugins/scm_filter/BitbucketAgedBranchesTrait.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import jenkins.scm.api.SCMSource;
+import jenkins.scm.api.trait.SCMSourceContext;
+import jenkins.scm.api.trait.SCMSourceRequest;
+import jenkins.scm.impl.trait.Selection;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.scm_filter.enums.RefType;
+import org.jenkinsci.plugins.scm_filter.utils.FilterRefUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
```
[the remark this answers is not in the document]
Comment (Trucosuso) [reply bitbucket-scm-filter-aged-refs/src/main/java/org/jenkinsci/plugins/scm_filter/BitbucketAgedBranchesTrait.java]: I removed all of the useless comments that were similar to the ones you mentioned

**Review by TobiX — APPROVED** [review]

**Pull request merged**